test/test_testing.py

```diff
@@ -77,21 +77,21 @@ def on_storage_changed(self, event: ops.EventBase):
         self.changes.append(event)
 
 
-class TestHarness(unittest.TestCase):
-    def test_add_relation_no_meta_fails(self):
+class TestHarness:
+    def test_add_relation_no_meta_fails(self, request: pytest.FixtureRequest):
         harness = ops.testing.Harness(ops.CharmBase, meta="name: mycharm")
-        self.addCleanup(harness.cleanup)
+        request.addfinalizer(harness.cleanup)
```

Comment:
I guess these can't use a `harness` fixture because all the tests have different metadata?

Reply:
Yes, and in previous draft PRs, passing parameters to fixtures has been proved to be not so beautiful, so here `request.addfinalizer` is used.